src/components/ui/toolbox.ts

```diff
@@ -262,24 +253,93 @@ export default class Toolbox extends EventsDispatcher<ToolboxEvent> {
   private get toolsToBeDisplayed(): BlockTool[] {
     return Array
       .from(this.tools.values())
-      .filter(tool => {
+      .reduce((result, tool) => {
         const toolToolboxSettings = tool.toolbox;
 
-        /**
-         * Skip tools that don't pass 'toolbox' property
-         */
-        if (!toolToolboxSettings) {
-          return false;
+        if (Array.isArray(toolToolboxSettings)) {
+          const validToolboxSettings = toolToolboxSettings
+            .filter(item => this.areToolboxSetttingsValid(item, tool.name))
+            .filter((item, i) => {
+              const notUnique = toolToolboxSettings.slice(0, i).find(otherItem => otherItem.id === item.id);
+
+              if (notUnique) {
+                _.log('Toolbox entry has not unique combination of icon and title. Toolbox entry %o is skipped', 'warn', item.title);
+
+                return false;
```

Comment:
maybe we should not skip non-unique items? I think, nothing horrible will happen, but the code will become a bit simpler